src/nvim/ex_cmds.c

```diff
@@ -4960,3 +4961,32 @@ void ex_oldfiles(exarg_T *eap)
     }
   }
 }
+
+void ex_trust(exarg_T *eap)
+{
+  char *p = skiptowhite(eap->arg);
+  size_t len = (size_t)(p - eap->arg);
+  char *arg1 = xmemdupz(eap->arg, len);
+
+  if (len > 2 && strcmp(arg1, "++deny") == 0) {
+    char *path = skipwhite(p);
+    if (path[0] == '\0') {
+      path = NULL;
+    }
+    nlua_trust("deny", path);
+  } else if (len > 2 && strcmp(arg1, "++remove") == 0) {
+    char *path = skipwhite(p);
+    if (path[0] == '\0') {
+      path = NULL;
+    }
+    nlua_trust("remove", path);
+  } else {
+    char *path = skipwhite(eap->arg);
+    if (path[0] == '\0') {
+      path = NULL;
+    }
+    nlua_trust("allow", path);
+  }
```

Comment:
This can be simplified to something like this
```suggestion
  const char *const p = skiptowhite(eap->arg);
  char *arg1 = xmemdupz(eap->arg, p - eap->arg);

  const char *action = "allow";
  const char *path = p;
  if (strcmp(arg1, "++deny") == 0) {
    path = skipwhite(p);
    action = "deny";
  } else if (strcmp(arg1, "++remove") == 0) {
    path = skipwhite(p);
    action = "remove";
  }
  if (path[0] == '\0') {
    path = NULL;
  }
  nlua_trust(action, path);
```

Reply:
Done slightly differently in 095df3056ef403f007ea33e49d30cb77f763ae10
I'm initializing `path` to `eap->arg`, so that when someone does `:trust ...` it throws an error, because it shouldn't have any arguments.